subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/SimpleConfiguration.kt

```diff
@@ -79,3 +83,22 @@ abstract class SimpleConfiguration(val module: InternalModule) : Equality {
     protected abstract fun containsBytecode(bytecodeDirectory: File): Boolean
     protected abstract fun containsSources(sourceSet: AndroidSourceSet): Boolean
 }
+
+internal fun SimpleConfiguration.dependencies(filter: (Configuration) -> Boolean): Set<ImplementationConfiguration> {
+    return module.project.configurations
+        .filter(filter)
+        .flatMap { configuration ->
+            configuration
+                .dependencies
+                .withType(DefaultProjectDependency::class.java)
+        }
+        .toSet()
+        .map {
+            it.dependencyProject
+                .internalModule
+                .implementationConfiguration
+        }
+        .toMutableSet().apply {
+            remove(this@dependencies)
```

Comment:
Somehow, in AGP 4.1 it found configuration itself. Whatever the reason, we don't need it in dependencies.